### PR TITLE
Handle additional hashing cases

### PR DESCRIPTION
**Pull request by carterkozak**
==COMMIT_MSG==
Handle additional hashing cases
==COMMIT_MSG==

### CODE REVIEW

File: baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/MoreAbstractAsKeyOfSetOrMap.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            Matchers.anyOf(GUAVA_CACHE_BUILDER, CAFFEINE_CACHE_BUILDER);
+
+    private static final Matcher<MethodInvocationTree> HASH_KEYED_COLLECTOR_METHODS = Matchers.anyOf(
+            SET_COLLECTOR,
+            UNMODIFIABLE_SET_COLLECTOR,
+            IMMUTABLE_SET_COLLECTOR,
+            MAP_COLLECTOR,
+            CONCURRENT_MAP_COLLECTOR,
+            UNMODIFIABLE_MAP_COLLECTOR,
+            IMMUTABLE_MAP_COLLECTOR);
```
Comment (schlosna) [inline baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/MoreAbstractAsKeyOfSetOrMap.java:91]: maybe overkill, but do we want to cover StreamEx `toImmutableMap` & `toImmutableSet` as well (`PreferImmutableStreamExCollections` pushes these when using `StreamEx`)?

**Review by schlosna — APPROVED** [review]

**Comment by schlosna** [comment]
@carterkozak defer to you on when you want to merge into the PR branch

File: baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/MoreAbstractAsKeyOfSetOrMap.java (hunk 1 above)
Comment (carterkozak) [reply baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/MoreAbstractAsKeyOfSetOrMap.java:91]: Let's punt those cases for now, I think they're less common than the stdlib/guava/cache cases, and I'm less familiar with StreamEx.

Comment (schlosna) [reply baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/MoreAbstractAsKeyOfSetOrMap.java:91]: Sounds good. We can also check the JDK 9+ `Map.of` & `Set.of` in a follow up .

**Pull request merged**